src/battle_script_commands.c

```diff
@@ -8300,6 +8300,40 @@ static void Cmd_various(void)
             gBattlescriptCurrInstr += 7;
         }
         return;
+    case VARIOUS_TOTEM_BOOST:
+        gActiveBattler = gBattlerAttacker;
+        if (gTotemBoosts[gActiveBattler].stats == 0)
+        {
+            gBattlescriptCurrInstr += 7;    //stats done, exit
+        }
+        else
+        {
+            for (i = 0; i < (NUM_BATTLE_STATS - 1); i++)
+            {
+                if (gTotemBoosts[gActiveBattler].stats & (1 << i))
+                {
+                    bool8 negative = (gTotemBoosts[gActiveBattler].statChanges[i] & 0x80) ? TRUE : FALSE;
+                    u8 change = gTotemBoosts[gActiveBattler].statChanges[i] & 0x7F;
```

Comment:
```suggestion
                    u8 change = gTotemBoosts[gActiveBattler].statChanges[i] & 0xF;
```
Is one solution to fixing an issue where numbers like -1 aren't properly being converted due to a cast from u8 to s8 in `ChangeStatBuff`.